test/parallel/test-policy-integrity.js

```diff
@@ -0,0 +1,287 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const tmpdir = require('../common/tmpdir');
+const assert = require('assert');
+const { spawnSync } = require('child_process');
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+const { pathToFileURL } = require('url');
+
+tmpdir.refresh();
+
+function hash(algo, body) {
+  const h = crypto.createHash(algo);
+  h.update(body);
+  return h.digest('base64');
+}
+
+const policyFilepath = path.join(tmpdir.path, 'policy');
+
+const packageFilepath = path.join(tmpdir.path, 'package.json');
+const packageURL = pathToFileURL(packageFilepath);
+const packageBody = '{"main": "dep.js"}';
+const policyToPackageRelativeURLString = `./${
+  path.relative(path.dirname(policyFilepath), packageFilepath)
+}`;
+
+const parentFilepath = path.join(tmpdir.path, 'parent.js');
+const parentURL = pathToFileURL(parentFilepath);
+const parentBody = 'require(\'./dep.js\')';
+
+const depFilepath = path.join(tmpdir.path, 'dep.js');
+const depURL = pathToFileURL(depFilepath);
+const depBody = '';
+const policyToDepRelativeURLString = `./${
+  path.relative(path.dirname(policyFilepath), depFilepath)
+}`;
+
+fs.writeFileSync(parentFilepath, parentBody);
+fs.writeFileSync(depFilepath, depBody);
+
+function test({
+  shouldFail = false,
+  entry,
+  onerror,
+  resources = {}
+}) {
+  const manifest = {
+    onerror,
+    resources: {}
+  };
+  for (const [url, { body, match }] of Object.entries(resources)) {
+    manifest.resources[url] = {
+      integrity: `sha256-${hash('sha256', match ? body : body + '\n')}`
+    };
+    fs.writeFileSync(new URL(url, pathToFileURL(tmpdir.path)), body);
+  }
+  fs.writeFileSync(policyFilepath, JSON.stringify(manifest, null, 2));
+  const { status } = spawnSync(process.execPath, [
+    '--experimental-policy', policyFilepath, entry
```

Comment:
Is it intended that multiple policy files are supported? If so can you add tests for cases where there are more than one policy files?

Reply:
Multiple could be supported in the future, right now it just uses 1 to avoid some injection questions. `NODE_OPTIONS` coming before `argv` leads us to use the last one specified particular so that CLI overtakes ENV. I could make it error in a follow on if desired.

Reply:
If so can you make the option a `std::string` instead of `std::vector<std::string>` in this patch?

Reply:
fixed